en/docs/develop/writing-a-custom-event-handler.md

```diff
@@ -1,228 +1,161 @@
-
-??? example "Click to view the sample listener implementation"
-    ``` java
-    public class SPWorkflowListener extends AbstractApplicationMgtListener {
-
-          @Override
-          public int getDefaultOrderId() {
-                  return 1;
-          }
-
-          @Override
-          public boolean doPreCreateApplication(ServiceProvider serviceProvider, String tenantDomain, String userName)
-                  throws IdentityApplicationManagementException {
-                  if (!isEnable()) {
-                          return true;
-                  }
-                  try {
-                          SPCreateHandler spCreateHandler = new SPCreateHandler();
-                          return spCreateHandler.startSPCreateWorkflow(serviceProvider, tenantDomain, userName);
-                  } catch (Exception e) {
-                          // Sending e.getMessage() since it is required to give error message to end user.
-                          throw new IdentityApplicationManagementException(e.getMessage(), e);
-                  } finally {
-                  }
-          }
+1. Override the `getName()` method to set the name for the event handler and the `getPriority()` method can be used to set the priory of the event handler. The handlers will be executed based on the priority.
 
+    ```
+    public String getName() {
+        return "customEvent";
+    }
+    @Override
+    public int getPriority(MessageContext messageContext) {
+        return 50;
     }
     ```
 
-Finally in the service component, we need to register the handler
-and the listener we implemented. We can do this as follows.
+3. To execute the expected operation, override the `handleEvent()` method. The `event.getEventProperties()` method can be used to get the parameters related to the user operations. 
+
+    ```
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+    Map<String, Object> eventProperties =event.getEventProperties();
+    String userName = (String) eventProperties.get(IdentityEventConstants.EventProperty.USER_NAME);
+    UserStoreManager userStoreManager = (UserStoreManager) eventProperties.get(IdentityEventConstants.EventProperty.USER_STORE_MANAGER);
+    String tenantDomain = (String) eventProperties.get(IdentityEventConstants.EventProperty.TENANT_DOMAIN);
+    String domainName = userStoreManager.getRealmConfiguration().getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
+    String eventName = event.getEventName();
+    if (IdentityEventConstants.Event.PRE_SET_EVENT.equals(eventName)) {//Do the operation on PRE_SET_EVENT.}
+    if(IdentityEventConstants.Event.POST_SET_EVENT.equals(eventName)) {//Do the operation on POST_SET_EVENT.}
+    }
+    ```
 
-``` java
-**
-* @scr.component name="tenant.mgt.workflow" immediate="true"
-*/
-public class SPWorkflowServiceComponent {
+## Using the event handler to perform an operation
```

Comment:
```suggestion
## Triggering a custom event and using the event handler to listen and perform an operation
```

Reply:
Isn't that a bit too long for a subheading? Any way we can make that shorter?

Reply:
This section needs to go at the end of the document, even after the part where we link available handlers.

Reply:
On second thought, let's remove the triggering events section which is irrelevant to writing an event handler.